src/transformers/trainer.py

```diff
@@ -1253,20 +1253,34 @@ def torch_jit_model_eval(self, model, dataloader, training=False):
                 return model
             jit_inputs = []
             example_batch = next(iter(dataloader))
+            example_batch = self._prepare_inputs(example_batch)
+            if self.label_smoother is not None and "labels" in example_batch:
+                example_batch.pop("labels")
             for key in example_batch:
                 example_tensor = torch.ones_like(example_batch[key])
                 jit_inputs.append(example_tensor)
             jit_inputs = tuple(jit_inputs)
             try:
                 jit_model = model.eval()
-                with ContextManagers([self.autocast_smart_context_manager(), torch.no_grad()]):
+                with ContextManagers(
```

Comment:
This is the logic that is in `self.autocast_smart_context_manager()` if I'm not mistaken so let's no change anything here.

Reply:
Hi, sgugger, the key point is we need set "cache_enabled=False" here, self.autocast_smart_context_manager() do not have suchargument setting. or else, torch+bf16/fp16+jit.trace will fail here.

Reply:
Hi, sgugger, the key point is we need set "cache_enabled=False" here, self.autocast_smart_context_manager() do not have such argument setting. or else, torch+bf16/fp16+jit.trace will fail here.

Reply:
Can we add that argument to the method then?